src/lib/first-hit-helpers.ts

```diff
@@ -154,10 +164,28 @@ export async function getStateFromUrl (url: URL): Promise<NavigationState> {
   const urlHasSubdomainConfigRequest = hasHashFragment(url, HASH_FRAGMENTS.IPFS_SW_SUBDOMAIN_REQUEST) && url.searchParams.get(QUERY_PARAMS.HELIA_SW) != null
   let hasConfig = false
   const supportsSubdomains = await checkSubdomainSupport(url)
+  let isHardRefresh = false
+  let isConfigRequest = false
 
   if (isIsolatedOrigin) {
     // check if indexedDb has config
     hasConfig = await isConfigSet(uiLogger)
+    if (hasConfig) {
+      // Check service worker state
+      const registration = await navigator.serviceWorker.getRegistration()
+      const hasActiveWorker = registration?.active != null
+      const hasControllingWorker = navigator.serviceWorker.controller != null
+
+      if (hasActiveWorker && !hasControllingWorker) {
```

Comment:
in chrome & firefox, hasActiveWorker will be true and hasControllingWorker will be false after a hard refresh.

in playwright-firefox, this is not true. it may be an out of date browser runtime?